### PR TITLE
[Feature #106009] Add support for recent DB-Dialects

### PR DESCRIPTION
**Pull request by MrEasy**

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java [hunk 1]
```diff
@@ -187,7 +187,6 @@ else if ( majorVersion == 10 && minorVersion == 5 ) {
 					}
 					else {
 						// assume `majorVersion > 13`
-						LOG.unknownSqlServerVersion( majorVersion, SQLServer2012Dialect.class );
```
Comment (ecki) [inline hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java:181]: hm yeah, would have added to case instead, but I guess this is less maintenance.

Comment (MrEasy) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java:181]: Took it from master where it is like this.

Comment (svogt) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java:181]: The switch itself didn't change anyway, just the log message has been removed, or?

**Review by svogt — APPROVED** [review]
Looks good to me though untested in a real BIS and with the respective databases...

File: hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java (hunk 1 above)
Comment (ecki) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java:181]: yes, but including case 15 in the switch wqould not log the default warning for 2019
but the outcome for 2019 is the same - no warning and using 2012 dialect.

Comment (ecki) [reply hibernate-core/src/main/java/org/hibernate/engine/jdbc/dialect/internal/StandardDialectResolver.java:181]: hm, I have actually approved the change, no idea why it dit not show up as approved.

**Review by ecki — APPROVED** [review]

**Pull request merged**